include/pmacc/memory/MakeUnique.hpp

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2019 Sergei Bastrakov
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+
+#include <memory>
+#include <utility>
+
+
+namespace pmacc
+{
+namespace memory
+{
+
+    /*
+     * Analogue of std::make_unique for C++11, except not disabled for arrays.
+     * Implementation is taken from
+     * https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique
+     */
+    template<
+        typename T,
+        typename ... T_Args
+    >
+    inline std::unique_ptr< T > makeUnique( T_Args && ... args )
```

Comment:
please use an anonymous namespace to support multiple compile units

Reply:
I think `inline` is fine here. [ODR](https://en.cppreference.com/w/cpp/language/definition) requires a single definition for non-inline functions, while for inline it is only required that each translation unit has a definition, and having a single definition overall is not required. Basically, that's what we use already with most methods implemented in .hpp files inside a class and thus automatically `inline`.

With the anonymous namespace, iirc, I suggested that at some point, but think here it is not a solution here. At best, it will work the same as inline with every translation unit having a separate, but same, definition. While if we had some internal state (e.g., static local variable inside), it would lead to errors or unexpected behaviour.